public/app/plugins/datasource/grafana-azure-monitor-datasource/module.ts

```diff
@@ -16,16 +17,41 @@ getAppEvents().subscribe<DashboardLoadedEvent<AzureMonitorQuery>>(
   DashboardLoadedEvent,
   ({ payload: { dashboardId, orgId, userId, grafanaVersion, queries } }) => {
     const azureQueries = queries[pluginJson.id];
+    let stats = {
+      [AzureQueryType.AzureMonitor]: {
+        hidden: 0,
+        visible: 0,
+      },
+      [AzureQueryType.LogAnalytics]: {
+        hidden: 0,
+        visible: 0,
+      },
+      [AzureQueryType.AzureResourceGraph]: {
+        hidden: 0,
+        visible: 0,
+      },
+    };
+    azureQueries.forEach((query) => {
+      if (
+        query.queryType === AzureQueryType.AzureMonitor ||
+        query.queryType === AzureQueryType.LogAnalytics ||
+        query.queryType === AzureQueryType.AzureResourceGraph
+      ) {
+        stats[query.queryType][query.hide ? 'hidden' : 'visible']++;
+      }
+    });
+
     if (azureQueries && azureQueries.length > 0) {
-      reportInteraction('grafana_ds_azuremonitor_dashboard_loaded', {
+      trackAzureMonitorDashboardLoaded({
         grafana_version: grafanaVersion,
         dashboard_id: dashboardId,
         org_id: orgId,
-        user_id: userId,
-        queries: azureQueries.map((query: AzureMonitorQuery) => ({
-          hidden: !!query.hide,
-          query_type: query.queryType,
-        })),
+        azure_monitor_queries: stats[AzureQueryType.AzureMonitor].visible,
+        azure_log_analytics_queries: stats[AzureQueryType.LogAnalytics].visible,
+        azure_resource_graph_queries: stats[AzureQueryType.AzureResourceGraph].visible,
+        azure_monitor_queries_hidden: stats[AzureQueryType.AzureMonitor].hidden,
```

Comment:
another nit: the distinction between visible and hidden seems a bit arbitrary. I know this is some simplified version of the query filtering but if we are just interested on visible queries, I would just record that.